plugins/wallabag.koplugin/main.lua

```diff
@@ -0,0 +1,695 @@
+                            num_deleted = num_deleted + 1
+                        end
+                    end
+                end -- has sidecar
+            end -- not . and ..
+        end -- for entry
+    end -- flag checks
+    return num_deleted
+end
+
+function Wallabag:deleteArticle( path )
+    logger.dbg("deleting article ", path )
+    local id = self:getArticleID( path )
+    if id then
+        self:callAPI( 'DELETE', "/api/entries/" .. id .. ".json", nil, "", "" )
+        self:deleteLocalArticle( path )
+    end
+end
+
+function Wallabag:deleteLocalArticle( path )
+    local entry_mode = lfs.attributes(path, "mode")
+    if entry_mode == "file" then
+        os.remove(path)
+        local sdr_dir = DocSettings:getSidecarDir( path )
+        FFIUtil.purgeDir( sdr_dir )
+        filemanagerutil.removeFileFromHistoryIfWanted( path )
+   end
+end
+
+function Wallabag:getArticleID( path )
+    -- extract the Wallabag ID from the file name
+    local offset = self.directory:len() + 2 -- skip / and advance to the next char
+    local preffix_len = article_id_preffix:len()
+    if path:sub( offset , offset + preffix_len - 1 ) ~= article_id_preffix then
+        logger.warn("getArticleID: no match! ", path:sub( offset , offset + preffix_len - 1 ) )
+        return
+    end
+    local endpos = path:find( article_id_postfix, offset + preffix_len )
+    if endpos == nil then
+        logger.warn("getArticleID: no match! " )
+        return
+    end
+    local id = path:sub( offset + preffix_len, endpos - 1 )
+    return id
+end
+
+function Wallabag:refreshCurrentDirIfNeeded()
+    if FileManager.instance then
+        FileManager.instance:onRefresh( )
+    end
+end
+
+function Wallabag:setFilterTag(touchmenu_instance)
+   self.tag_dialog = InputDialog:new {
+        title =  _("Set a single tag to filter articles on"),
+        input = self.filter_tag,
+        input_type = "string",
+        buttons = {
+            {
+                {
+                    text = _("Cancel"),
+                    callback = function()
+                        UIManager:close(self.tag_dialog)
+                    end,
+                },
+                {
+                    text = _("OK"),
+                    is_enter_default = true,
+                    callback = function()
+                        self.filter_tag = self.tag_dialog:getInputText()
+                        self:saveSettings()
+                        touchmenu_instance:updateItems()
+                        UIManager:close(self.tag_dialog)
+                    end,
+                }
+            }
+        },
+    }
+    UIManager:show(self.tag_dialog)
+    self.tag_dialog:onShowKeyboard()
+end
+
+function Wallabag:editServerSettings()
+    local text_info = [[Enter the details of your Wallabag server and account.
+
+Client ID and client secret are long strings so you might prefer to save the empty settings and edit the config file directly:
+.adds/koreader/settings/wallabag.lua
```

Comment:
This is Kobo-specific (and even then mainly for KSM), plus I suspect you typod settings? You should populate this with `DataStorage:getSettingsDir()`.

https://github.com/koreader/koreader/blob/fa0117bb2a02d0984bded06144c93218a85b21cb/datastorage.lua#L36